pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -19,12 +19,175 @@ package kuberuntime
 import (
 	"fmt"
 	"io"
+	"os"
+	"path"
 
+	"github.com/golang/glog"
 	"k8s.io/kubernetes/pkg/api"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/types"
 	"k8s.io/kubernetes/pkg/util/term"
 )
 
+// getContainerLogsPath gets log path for container.
+func getContainerLogsPath(containerName string, podUID types.UID) string {
+	return path.Join(podLogsRootDirectory, string(podUID), fmt.Sprintf("%s.log", containerName))
+}
+
+// generateContainerConfig generates container config for kubelet runtime api.
+func (m *kubeGenericRuntimeManager) generateContainerConfig(container *api.Container, pod *api.Pod, restartCount int, podIP string) (*runtimeApi.ContainerConfig, error) {
+	opts, err := m.runtimeHelper.GenerateRunContainerOptions(pod, container, podIP)
+	if err != nil {
+		return nil, err
+	}
+
+	_, containerName, cid := buildContainerName(pod, container)
```

Comment:
@yujuhong @feiskyer @Random-Liu @euank So are we going to use the container name to store all those info? It seems this will turn out to be an implicit contract between runtime and kubelet?
IIUC, the name is contructed  this way for docker so that kubelet can list the containers that are managed by kubelet. But that's not the only way to solve this problem, and I don't feel the shared runtime code needs to care about this.

The runtime itself should be responsible to insert the `container name/pod name/pod namespace` somewhere, either in the actual container name, or annotation, etc. And when kubelet ask list containers/list pods, the runtime should be able to retrieve the kubelet-managed containers.